src/odemis/driver/lakeshore.py

```diff
@@ -163,45 +220,72 @@ def _openSerialPort(port, baudrate=57600):
 
         return ser
 
-    def _findDevice(self, ports):
+    def _checkDeviceCompatibility(self):
+        """
+        Checks the device manifacturer and model
+        raises:
+         IOError: Not supported device model or manifacturer
+         There might be other errors such as timeout error
+        """
+        manufacturer, md, _, _ = self.GetIdentifier()  # if value is incorrect, will throw an exception wile unpacking
+
+        if manufacturer != "LSCI":
+            raise IOError("Invalid device manufacturer")
+
+        compatible_devices = ["MODEL335", "MODEL350"]
+        if md not in compatible_devices:
+            raise IOError("Incompatible device model detected. "
+                          "Compatible models are %s." % (compatible_devices,))
+
+    def _findDevice(self, bus_pattern):
         """
         Look for a compatible device
-        ports (str): pattern for the port name
+        bus_pattern (str): pattern for the port name
```

Comment:
Update the docstring. Indicate that if it doesn't start with a "/" then it should be an IP address.